crates/node-db/tests/insert.rs

```diff
@@ -17,13 +17,17 @@ fn test_insert_block() {
     // Create an in-memory SQLite database
     let mut conn = test_conn();
 
-    // Create the necessary tables and insert the block.
-    let tx = conn.transaction().unwrap();
-    node_db::create_tables(&tx).unwrap();
-    for block in &blocks {
-        node_db::insert_block(&tx, block).unwrap();
-    }
-    tx.commit().unwrap();
+    let mut fetched_blocks = vec![];
+    node_db::with_tx::<_, QueryError>(&mut conn, |tx| {
+        // Create the necessary tables and insert blocks
+        node_db::create_tables(tx).unwrap();
+        for block in &blocks {
+            node_db::insert_block(tx, block).unwrap();
+        }
+        fetched_blocks = node_db::list_blocks(tx, 0..100).unwrap();
+        Ok(())
+    })
+    .unwrap();
```

Comment:
An alternative could be to return the listed blocks from the fn:

```suggestion
    let fetched_blocks = node_db::with_tx(&mut conn, |tx| {
        // Create the necessary tables and insert blocks
        node_db::create_tables(tx).unwrap();
        for block in &blocks {
            node_db::insert_block(tx, block).unwrap();
        }
        node_db::list_blocks(tx, 0..100)
    })
    .unwrap();
```
There might be some other locations where we want to do this too

Reply:
Yeah, definitely cleaner.

Reply:
I see that the call to list_blocks is not actually needed here, I must have introduced that by mistake. 
That gets rid of the need for the `fetched_blocks` var, but as the `for` loop returns the unit type, I think I still need to unwrap and return `Ok(())`.

Will happily change if there's a cleaner/more idiomatic way here?